odoo/fields.py

```diff
@@ -1711,7 +1726,7 @@ def convert_to_write(self, value, record):
         return value
 
 
-class _String(Field):
+class _String(Field[str]):
```

Comment:
```suggestion
class _String(Field[str | typing.Literal[False]]):
```

same for dates and selection.